common/expirecontainer.h

```diff
@@ -372,3 +373,124 @@ class ObjectCache : public ObjectCacheBase {
         return borrow(key, [] { return new ValEntity(); });
     }
 };
+
+template <typename KeyType, typename NodeType>
+class ObjectCache<KeyType, intrusive_list<NodeType>> : public ObjectCacheBase {
+protected:
+    using ListType = intrusive_list<NodeType>;
+    using Base = ObjectCacheBase;
+    using KeyedItem = Base::KeyedItem<Base::Item, KeyType>;
+
+    class Item : public KeyedItem {
+    public:
+        ListType _list;
+        using KeyedItem::KeyedItem;
+        virtual Item* construct() const override {
+            auto item = new Item(this->_key);
+            item->_obj = nullptr;
```

Comment:
```item->_obj = item;``` ?

Reply:
the `construct` member function should keep _obj as nullptr.

In `ref_acquire` implementation, item will create if not exists, and then check if _obj is empty. The real `ctor` will be called only when `_obj` equals to nullptr